Can not compile with clang 16 and libc++

**Describe your environment** 

OS: Linux
Compiler: Clang 16.0.1 with `-stdlib=libc++`
Build system: cmake
otel-cpp version: v1.9.0

**Steps to reproduce**

Just compile otel-cpp.

**What is the expected behavior?**
Success

**What is the actual behavior?**

```bash
[ 61%] Building CXX object ext/src/http/client/curl/CMakeFiles/opentelemetry_http_client_curl.dir/http_client_factory_curl.cc.o
In file included from ~/opentelemetry-cpp-v1.9.0/ext/src/http/client/curl/http_client_factory_curl.cc:4:
In file included from ~/opentelemetry-cpp-v1.9.0/ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h:6:
~/opentelemetry-cpp-v1.9.0/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h:185:45: error: no template named 'function' in namespace 'std'
  CURLcode SendAsync(Session *session, std::function<void(HttpOperation &)> callback = nullptr);
                                       ~~~~~^
~/opentelemetry-cpp-v1.9.0/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h:310:10: error: no template named 'function' in namespace 'std'
    std::function<void(HttpOperation &)> callback;
    ~~~~~^
~/opentelemetry-cpp-v1.9.0/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h:185:77: error: cannot initialize a parameter of type 'int' with an rvalue of type 'std::nullptr_t'
  CURLcode SendAsync(Session *session, std::function<void(HttpOperation &)> callback = nullptr);
                                                                            ^          ~~~~~~~
~/opentelemetry-cpp-v1.9.0/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h:185:77: note: passing argument to parameter 'callback' here
3 errors generated.
gmake[2]: *** [ext/src/http/client/curl/CMakeFiles/opentelemetry_http_client_curl.dir/build.make:76: ext/src/http/client/curl/CMakeFiles/opentelemetry_http_client_curl.dir/http_client_factory_curl.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:692: ext/src/http/client/curl/CMakeFiles/opentelemetry_http_client_curl.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
'/opt/cmake/bin/cmake' '--install' '.' '--prefix' '/home/owent/workspace/tencent/opentelemetry-cpp-ecosystem/third_party/install/linux-x86_64-clang'
-- Install configuration: "RelWithDebInfo"
```

--- a/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h
+++ b/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h
@@ -13,6 +13,7 @@
 #  include <future>
 #endif
 
+#include <functional>
 #include <map>
 #include <regex>
 #include <sstream>

